math/mode.go

```diff
@@ -0,0 +1,61 @@
+// mode.go
+// author(s): [CalvinNJK] (https://github.com/CalvinNJK)
+// description: Finding Mode Value In an Array
+// see mode.go
+
+package math
+
+//package constraints
+
+import (
+	"errors"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+type Number interface {
+	constraints.Float | constraints.Integer
+}
```

Comment:
We already have [`constraints.Number`](https://github.com/TheAlgorithms/Go/blob/master/constraints/contraints.go#L28-L30)